heron/tmaster/src/cpp/manager/tcontroller.cpp

```diff
@@ -211,5 +192,125 @@ void TController::HandleCleanStatefulCheckpointRequestDone(IncomingHTTPRequest*
   }
   delete request;
 }
+
+void TController::HandleRuntimeConfigRequest(IncomingHTTPRequest* request) {
+  LOG(INFO) << "Got a RuntimeConfig request from " << request->GetRemoteHost() << ":"
+            << request->GetRemotePort();
+  ValidationResult result;
+  if (!ValidateTopology(request, result)) {
+    http_server_->SendErrorReply(request, result.GetCode(), result.GetMessage());
+    delete request;
+    return;
+  }
+
+    // Look for runtime-config parameters
+  std::vector<sp_string> parameters;
+  if (!request->GetAllValues("runtime-config", parameters)) {
+    LOG(ERROR) << "No runtime config is found";
+    http_server_->SendErrorReply(request, 400, "No runtime config is found."
+        " Usage: runtime-config=[COMPONENT:]<CONFIG>:<VALUE>.");
+    delete request;
+    return;
+  }
+  LOG(INFO) << "Found " << parameters.size() << " configs in request.";
+
+  // Parse new configs in request
+  std::map<std::string, std::map<std::string, std::string>> config;
+  if (!ParseRuntimeConfig(parameters, config)) {
+    http_server_->SendErrorReply(request, 400, "Failed to parse runtime configs."
+        " Possibly bad format. The expected format is [COMPONENT:]<CONFIG>:<VALUE>.");
+    delete request;
+    return;
+  }
+
+  // Validate them before applying
+  if (!tmaster_->ValidateRuntimeConfig(config)) {
+    http_server_->SendErrorReply(request, 400, "Failed to validate runtime configs");
+    delete request;
+    return;
+  }
+
+  auto cb = [request, this](proto::system::StatusCode status) {
+    this->HandleRuntimeConfigRequestDone(request, status);
+  };
+
+  if (!tmaster_->RuntimeConfigTopology(config, std::move(cb))) {
```

Comment:
tmaster_->UpdateRuntimeConfigTopology

Reply:
I am going to use UpdateRuntimeConfig for this one.